wolfcrypt/src/port/nxp/se050_port.c

```diff
@@ -0,0 +1,1418 @@
+/* se050_port.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <stdint.h>
+
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/wolfcrypt/types.h>
+#include <wolfssl/wolfcrypt/wc_port.h>
+#include <wolfssl/wolfcrypt/aes.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+
+
+#if defined(WOLFSSL_SE050)
+
+#include <wolfssl/wolfcrypt/port/nxp/se050_port.h>
+#include "fsl_sss_api.h"
+/*#include "fsl_sss_se05x_apis.c"
+#include <se05x_APDU.h>
+#include <se05x_tlv.h>*/
+
+
+//#include "se05x_APDU_apis.h"
+//#include "fsl_sss_se05x_apis.h"
+//Se05x_API_WriteECKey
+
+
+static sss_session_t *cfg_se050_i2c_pi;
+static sss_key_store_t *hostSession;
+int keyId_allocater = 100;
+
+
+int wolfcrypt_se050_SetConfig(sss_session_t *pSession, sss_key_store_t *pHostSession) 
+{
+	printf("Setting SE050 session configuration\n");
+	
+	XMEMSET(&cfg_se050_i2c_pi, 0, sizeof(cfg_se050_i2c_pi));
+	cfg_se050_i2c_pi = pSession;
+	
+    XMEMSET(&hostSession, 0, sizeof(hostSession));
+	hostSession = pHostSession;
+    
+	return 0;
+}
+
+int se050_allocate_key()
+{
+    //keyId_allocater = rand();
+    //return keyId_allocater;
+    keyId_allocater++;
+    return keyId_allocater++;
+}
+
+
+int se050_get_random_number(uint32_t count, uint8_t* rand_out)
+{
+    sss_status_t status;
+	sss_rng_context_t rng;
+
+
+    if (wolfSSL_CryptHwMutexLock() == 0) { 
+        status = sss_rng_context_init(&rng, cfg_se050_i2c_pi);
+        if (status != kStatus_SSS_Success) {
+            return -1;
```

Comment:
This will not unlock HW mutex. Please cleanup to have a single return point so `wolfSSL_CryptHwMutexUnLock` is always called.